src/search/pdbs/pattern_generator.h

```diff
@@ -6,24 +6,49 @@
 #include "types.h"
 
 #include <memory>
```

Comment:
Might not be related to this issue, but as we take a look currently at those generators:

The code for path the Pattern and the PatternCollection generators is almost a perfect copy. If this is expected to stay, the code can be simplified using templates.

1. create a templated generator : `template<typename T> Generator{...; virtual T compute_patterns(..) = 0;
2. class PatternGenerator : public Generator<PatternInformation>{};
3. class PatternCollectionGenerator : public Generator<PatternCollectionInformation>{};

The code in the cc files is also an almost perfect copy. The only difference are the dump_pattern(_collection)_statistics, but this can be easily done with overloading.

Is there a reason to have those duplication?

Reply:
I don't know if there is a good reason other than not wanting to use templates. I'm somewhat indifferent on this, maybe someone else has an opinion. Can you ask on Discord?